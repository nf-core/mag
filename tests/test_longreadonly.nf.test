nextflow_pipeline {

    name "Test pipeline with test_longreadonly profile"
    script "../main.nf"
    tag "pipeline"
    tag "test_longreadonly"
    profile "test_longreadonly"

    test("-profile longreadonly") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {

            // Output directory: multiqc
            def stable_name_multiqc = getAllFilesFromDir(
                params.outdir, include: ['multiqc/**'], relative: true, includeDir: false
            )
            def stable_path_multiqc = getAllFilesFromDir(
                params.outdir, include: ['multiqc/**'], ignoreFile: 'tests/.nftignore'
            )

            assert workflow.success
            assertAll(
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we test pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_mag_software_mqc_versions.yml"),
                ).match() },
                { assert snapshot(stable_name_multiqc, stable_path_multiqc).match('multiqc') },
            )
        }
    }
}
