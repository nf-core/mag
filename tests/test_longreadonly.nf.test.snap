{
    "-profile longreadonly": {
        "content": [
            57,
            {
                "ADJUST_MAXBIN2_EXT": {
                    "coreutils": 9.5
                },
                "BIN_SUMMARY": {
                    "python": "3.10.6",
                    "pandas": "1.4.3"
                },
                "BUSCO_BUSCO": {
                    "busco": "6.0.0"
                },
                "BUSCO_UNTAR": {
                    "untar": 1.34
                },
                "CONCAT_BINQC_TSV": {
                    "csvtk": "0.31.0"
                },
                "CONVERT_DEPTHS": {
                    "bioawk": 20110810
                },
                "FILTLONG": {
                    "filtlong": "0.2.1"
                },
                "FLYE": {
                    "flye": "2.9.5-b1801"
                },
                "GUNZIP_BINS": {
                    "gunzip": 1.13
                },
                "GUNZIP_LONGREAD_ASSEMBLIES": {
                    "gunzip": 1.13
                },
                "MAG_DEPTHS": {
                    "python": "3.6.7",
                    "pandas": "1.1.5"
                },
                "MAG_DEPTHS_SUMMARY": {
                    "python": "3.10.6",
                    "pandas": "1.4.3"
                },
                "MAXBIN2": {
                    "maxbin2": "2.2.7"
                },
                "METABAT2_JGISUMMARIZEBAMCONTIGDEPTHS_LONGREAD": {
                    "metabat2": 2.15
                },
                "METABAT2_METABAT2": {
                    "metabat2": 2.17
                },
                "METAMDBG_ASM": {
                    "metamdbg": 1.1
                },
                "MINIMAP2_ASSEMBLY_ALIGN": {
                    "minimap2": "2.29-r1283",
                    "samtools": 1.21
                },
                "MINIMAP2_ASSEMBLY_INDEX": {
                    "minimap2": "2.29-r1283"
                },
                "MINIMAP2_HOST_ALIGN": {
                    "minimap2": "2.29-r1283",
                    "samtools": 1.21
                },
                "MINIMAP2_HOST_INDEX": {
                    "minimap2": "2.29-r1283"
                },
                "NANOLYSE": {
                    "nanolyse": "1.2.0"
                },
                "NANOPLOT_FILTERED": {
                    "nanoplot": "1.46.1"
                },
                "NANOPLOT_RAW": {
                    "nanoplot": "1.46.1"
                },
                "PORECHOP_ABI": {
                    "porechop_abi": "0.5.0"
                },
                "PRODIGAL": {
                    "prodigal": "2.6.3",
                    "pigz": 2.6
                },
                "PROKKA": {
                    "prokka": "1.14.6"
                },
                "QUAST": {
                    "python": "3.7.6",
                    "metaquast": "5.0.2"
                },
                "QUAST_BINS": {
                    "python": "3.7.6",
                    "metaquast": "5.0.2"
                },
                "QUAST_BINS_SUMMARY": {
                    "cp": 9.5
                },
                "SAMTOOLS_HOSTREMOVED_INDEX": {
                    "samtools": 1.21
                },
                "SAMTOOLS_HOSTREMOVED_STATS": {
                    "samtools": 1.21
                },
                "SAMTOOLS_HOSTREMOVED_UNMAPPED": {
                    "samtools": 1.21
                },
                "SEQKIT_STATS": {
                    "seqkit": "2.9.0"
                },
                "SPLIT_FASTA": {
                    "python": "3.6.7",
                    "biopython": "1.7.4",
                    "pandas": "1.1.5"
                },
                "Workflow": {
                    "nf-core/mag": "v5.1.0dev"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-23T05:39:42.646278917"
    },
    "multiqc": {
        "content": [
            [
                "multiqc/multiqc_data/busco_plot_bacteria_odb10.yaml",
                "multiqc/multiqc_data/llms-full.txt",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc.parquet",
                "multiqc/multiqc_data/multiqc_busco.yaml",
                "multiqc/multiqc_data/multiqc_citations.yaml",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_general_stats.yaml",
                "multiqc/multiqc_data/multiqc_prokka.yaml",
                "multiqc/multiqc_data/multiqc_quast.yaml",
                "multiqc/multiqc_data/multiqc_quast_quast-1.yaml",
                "multiqc/multiqc_data/multiqc_software_versions.yaml",
                "multiqc/multiqc_data/multiqc_sources.yaml",
                "multiqc/multiqc_data/porechop.yaml",
                "multiqc/multiqc_data/prokka_plot.yaml",
                "multiqc/multiqc_data/quast_num_contigs.yaml",
                "multiqc/multiqc_data/quast_table.yaml",
                "multiqc/multiqc_plots/pdf/busco_plot_bacteria_odb10-cnt.pdf",
                "multiqc/multiqc_plots/pdf/busco_plot_bacteria_odb10-pct.pdf",
                "multiqc/multiqc_plots/pdf/prokka_plot-cnt.pdf",
                "multiqc/multiqc_plots/pdf/prokka_plot-pct.pdf",
                "multiqc/multiqc_plots/pdf/quast_num_contigs-cnt.pdf",
                "multiqc/multiqc_plots/pdf/quast_num_contigs-pct.pdf",
                "multiqc/multiqc_plots/pdf/quast_table.pdf",
                "multiqc/multiqc_plots/png/busco_plot_bacteria_odb10-cnt.png",
                "multiqc/multiqc_plots/png/busco_plot_bacteria_odb10-pct.png",
                "multiqc/multiqc_plots/png/prokka_plot-cnt.png",
                "multiqc/multiqc_plots/png/prokka_plot-pct.png",
                "multiqc/multiqc_plots/png/quast_num_contigs-cnt.png",
                "multiqc/multiqc_plots/png/quast_num_contigs-pct.png",
                "multiqc/multiqc_plots/png/quast_table.png",
                "multiqc/multiqc_plots/svg/busco_plot_bacteria_odb10-cnt.svg",
                "multiqc/multiqc_plots/svg/busco_plot_bacteria_odb10-pct.svg",
                "multiqc/multiqc_plots/svg/prokka_plot-cnt.svg",
                "multiqc/multiqc_plots/svg/prokka_plot-pct.svg",
                "multiqc/multiqc_plots/svg/quast_num_contigs-cnt.svg",
                "multiqc/multiqc_plots/svg/quast_num_contigs-pct.svg",
                "multiqc/multiqc_plots/svg/quast_table.svg",
                "multiqc/multiqc_report.html"
            ],
            [
                
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-10-17T15:53:16.48624226"
    }
}